packages/graphback-codegen-client/src/templates/gqlTemplates.ts

```diff
@@ -7,10 +7,22 @@ export const variableFields = (t: InputModelTypeContext) => {
     .join(', ')
 }
 
+export const generateInputVariableFields = (t: InputModelTypeContext, relatedTypes: InputModelTypeContext[]) => {
+  return t.fields.filter((f: InputModelFieldContext) => !f.isArray && f.type !== 'ID')
+    .map((f: InputModelFieldContext) => {
 
-export const inputVariableFields = (t: InputModelTypeContext) => {
-  return t.fields.filter((f: InputModelFieldContext) => !f.isType && !f.isArray && f.type !== 'ID')
-    .map((f: InputModelFieldContext) => `\$${f.name}: ${f.type}${f.isNull ? '' : '!'}`)
+      if (f.isType) {
+        const relation = relatedTypes.find((r: InputModelTypeContext) => r.name === f.name);
+
+        // TODO: get the relationship constraint type and name
```

Comment:
Next we would find the type that of the primary key of the related model and use that in the mutation